uefi/src/proto/device_path/mod.rs

```diff
@@ -1017,4 +1055,28 @@ mod tests {
         raw_data[2] += 1;
         assert!(<&DevicePathNode>::try_from(raw_data.as_slice()).is_err());
     }
+
+    #[test]
```

Comment:
99% of this is copied from `test_device_path_nodes`, should I put the shared code into one func?

Reply:
I'm fine with keeping the test code somewhat verbose so that it's easier to see exactly what it's testing.